distributed/core.py

```diff
@@ -49,6 +49,9 @@ def get_total_physical_memory():
 MAX_BUFFER_SIZE = get_total_physical_memory()
 
 
+CommunicationErrors = (StreamClosedError, socket.error, IOError, RPCClosed)
```

Comment:
RPCClosed ends up being convenient.  I don't now recall the original motivation.  I suspect that it was difficult to always ensure that the system hadn't closed down in between every yield point and using exceptions was simpler.